### PR TITLE
Redis SSL Support [#556]

### LINKED ISSUE #556: Add support for Redis SSL connections to Elasticache (r
### Is your feature request related to a specific problem?

In many production deployments, it's useful to enable Redis SSL (also referred to as Redis TLS, see a long Internet history of using SSL/TLS interchangeably 😄). I suspect the most common use case for this will be using AWS Elasticache, which is AWS' hosted version of Redis (they call it Elasticache to distance from Redis the company). Elasticache supports the ability to enable TLS encryption by default, docs for that are here: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html

Anyways, Redis SSL can be a bit complicated because there are many ways to do the certificate exchange, including:
1. Use root CA-signed (Lets Encrypt, ACM, etc) certificates: this is the simplest way, because most Redis clients should be able to verify and connect natively without having to provide any kind of certificate files to the client
2. Use self-signed certificates: this avoids paying for root CA-signed certificates, but requires passing extra params to the client like a `ssl_ca_certs` path to a certificate chain, etc.
3. Use self-signed certificates with client-side certificate authentication: this complicates it even further by requiring the client to also have it's own certificate that the server can validate

Because there are a lot of options here, you'll find a lot of confusing documentation online. However, all I *really* think we need to support now is option 1, because that's what Elasticache uses.

You can see some basic documentation for enabling SSL connections in `redis-py` here: https://github.com/redis/redis-py#ssl-connections.

### Describe the solution you'd like

I think we should essentially just expose the various connection options to `redis-py` as documented above. In particular, I think we should at least include the `ssl` and `ssl_cert_reqs` options used in the third example (disabling hostname verification), because often in an Elasticache deployment there may be a known mismatch between the hostname and the certificate domain because the implementer may setup a friendly CNAME (e.g. `fidesops.redis.mycompany.com`) to redirect to the FQDN of the Elasticache instance (e.g. `master.**********.cache.amazonaws.com`). This is a security vulnerability so we shouldn't disable hostname verification by default of course, but I suspect we'll have several users who know the risk and still want to do that for good reasons. See https://github.com/redis/redis-py/issues/1080 for several examples of why users might need that option!

In other words, I think if we had some config vars like `REDIS_SSL` and `REDIS_SSL_CERT_REQS` that were passed along to the `redis-py` library, we'd be pretty future proof for most Redis SSL connections.

### Describe alternatives you've considered, if any

n/a

### PR DESCRIPTION
**Pull request by pattisdr**
# Purpose
Add support for Redis SSL connections to Elasticache.

# Changes
Exposes simple config variables to enable Redis SSL connections.  Testing isn't included here - I think we can wait until we are hosting an instance of Fidesops that uses Elasticache.  

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [ ] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [ ] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [ ] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #556

### CODE REVIEW

File: fidesops.toml [hunk 1]
```diff
@@ -16,6 +16,8 @@ CHARSET = "utf8"
 DEFAULT_TTL_SECONDS = 604800
 DB_INDEX = 0
 ENABLED = true
+SSL = false
+SSL_CERT_REQS = "required"
```
Comment (sanders41) [inline fidesops.toml:20]: Is it correct for this to be set to `required` when `SSL = false`?

File: src/fidesops/core/config.py [hunk 2]
```diff
@@ -131,6 +131,8 @@ class RedisSettings(FidesSettings):
     DEFAULT_TTL_SECONDS: int = 604800
     DB_INDEX: int
     ENABLED: bool = True
+    SSL: bool = False
+    SSL_CERT_REQS: Optional[str] = "required"
```
Comment (sanders41) [inline src/fidesops/core/config.py:135]: Should this default to `None`?

File: fidesops.toml (hunk 1 above)
Comment (sanders41) [reply fidesops.toml:20]: Looks like you are correct, seems like the right thing to do.

**Comment by pattisdr** [comment]
updated, thanks @sanders41

**Pull request merged**